src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -175,6 +182,13 @@ const SignOrExecuteForm = ({
           willRelay={willRelay}
         />
 
+        <ExecutionMethodSelector
+          executionMethod={executionMethod}
+          setExecutionMethod={setExecutionMethod}
+          canRelay={canRelay}
```

Comment:
This prop is only being used for the rendering logic of `ExecutionMethodSelector`. I would suggest we instead check it here and not pass it further down e.g.

```
{canRelay && <ExecutionMethodSelector ...>}